dist/valid-head.js

```diff
@@ -0,0 +1,23 @@
+// Used for WPT
+// [valid-head]
+// Use DOMParser to extract head elements so browser can't truncate the head
+// Loop through each node and check against valid elements array
+// Returns object with ARRAY and BOOL
+var html = $WPT_BODIES[0].response_body;
+var parser = new DOMParser();
+var xmlDoc = parser.parseFromString(html,"text/xml");
```

Comment:
Oh interesting. How different is this approach from `createHTMLDocument`? See the wpt_bodies custom metric for example:

https://github.com/HTTPArchive/custom-metrics/blob/4c6bc0493b1b51dc021d6dcf7471b80224dc9228/dist/wpt_bodies.js#L82-L84

Is parsing as XML somehow more strict? Or for the purposes of validating the head, is it actually necessary to parse as XML?

Reply:
So XML allows me to extract the head elements as they are sent. The createHTMLDocument truncates the document just like the browser would. I ran some additional tests and my function returns the number of element nodes correctly. However, once parsed as XML, it is NOT valid HTML anymore.